src/webgpu/shader/validation/expression/call/builtin/bitcast.spec.ts

```diff
@@ -0,0 +1,141 @@
+export const description = `
+Validation negative tests for bitcast builtins.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { kBit } from '../../../../../util/constants.js';
+import { linearRange } from '../../../../../util/math.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+// A VectorCase specfies the number of components a vector type has,
+// and which component will have a bad value.
+// Use width = 1 to indicate a scalar.
+type VectorCase = { width: number; badIndex: number };
+function vectorCases(): VectorCase[] {
```

Comment:
nit: Can this be a const value instead of a function?
Something like this:
```
const vectorCases: VectorCase[] = {
    { width: 1, badIndex: 0 },
    { width: 2, badIndex: 0 },
    { width: 2, badIndex: 1 },
    { width: 3, badIndex: 0 },
    { width: 3, badIndex: 1 },
    { width: 3, badIndex: 2 },
    { width: 4, badIndex: 0 },
    { width: 4, badIndex: 1 },
    { width: 4, badIndex: 2 },
    { width: 4, badIndex: 3 },
};
```